aiowebostv/webos_client.py

```diff
@@ -133,238 +135,258 @@ async def close_client_session(self) -> None:
-
-            closeout = set()
-            closeout.update(handler_tasks)
-
-            if main_ws is not None:
-                closeout.add(asyncio.create_task(main_ws.close()))
-            if input_ws is not None:
-                closeout.add(asyncio.create_task(input_ws.close()))
-            if self.created_client_session:
-                closeout.add(asyncio.create_task(self.close_client_session()))
-
-            self.connection = None
-            self.input_connection = None
-
-            self.do_state_update = False
-
-            self._power_state = {}
-            self._current_app_id = None
-            self._muted = None
-            self._volume = None
-            self._current_channel = None
-            self._channel_info = None
-            self._channels = None
-            self._apps = {}
-            self._extinputs = {}
-            self._system_info = {}
-            self._software_info = {}
-            self._hello_info = {}
-            self._sound_output = None
-            self._media_state = []
+        Avoid partial updates during initial subscription.
+        """
+        self.do_state_update = False
+        self._system_info, self._software_info = await asyncio.gather(
+            self.get_system_info(), self.get_software_info()
+        )
+        subscribe_state_updates = {
+            self.subscribe_power_state(self.set_power_state),
+            self.subscribe_current_app(self.set_current_app_state),
+            self.subscribe_muted(self.set_muted_state),
+            self.subscribe_volume(self.set_volume_state),
+            self.subscribe_apps(self.set_apps_state),
+            self.subscribe_inputs(self.set_inputs_state),
+            self.subscribe_sound_output(self.set_sound_output_state),
+            self.subscribe_media_foreground_app(self.set_media_state),
+        }
+        subscribe_tasks = set()
+        for state_update in subscribe_state_updates:
+            subscribe_tasks.add(asyncio.create_task(state_update))
+        await asyncio.wait(subscribe_tasks)
+        for task in subscribe_tasks:
+            with suppress(WebOsTvServiceNotFoundError):
+                task.result()
+        # set placeholder power state if not available
+        if not self._power_state:
+            self._power_state = {"state": "Unknown"}
+        self.do_state_update = True
+        if self.state_update_callbacks:
+            await self.do_state_update_callbacks()
+
+    def _clear_tv_states(self) -> None:
+        """Clear all TV states."""
+        self._power_state = {}
+        self._current_app_id = None
+        self._muted = None
+        self._volume = None
+        self._current_channel = None
+        self._channel_info = None
+        self._channels = None
+        self._apps = {}
+        self._extinputs = {}
+        self._system_info = {}
+        self._software_info = {}
+        self._hello_info = {}
+        self._sound_output = None
+        self._media_state = []
```

Comment:
In the future, I'd consider making the state a separate object so you have a bit more separation between what the client is doing and what the state of the device is, and so you could call `self._state.clear()`

Reply:
That is my plan